### PR TITLE
Add ShuffleNetV2

### PR DESCRIPTION
**Pull request by vballoli**
This model introduces channel shuffling, so I thought it would be a good addition to the classification family.

### CODE REVIEW

File: Models/ImageClassification/ShuffleNetV2.swift [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+                padding: .valid
+            )
+            BatchNorm<Float>(featureCount: filters.0)
+            Conv2D<Float>(
+                filterShape: (1, 1, filters.0, branchChannels), strides: (1, 1), padding: .valid,
+                useBias: false
+            )
+            BatchNorm<Float>(featureCount: branchChannels)
+        }
+        var inputChannels = branchChannels
```
Comment (xihui-wu) [inline Models/ImageClassification/ShuffleNetV2.swift:73]: nit: can squeeze into one row. inputChannels = includeBranch ? filters.0: branchChannels

Comment (vballoli) [reply Models/ImageClassification/ShuffleNetV2.swift:73]: Oh yeah, that's better. I'll push the changes in sometime.

**Comment by vballoli** [comment]
Any particular training/inference code you have in mind for specific datasets, etc ?

**Comment by xihui-wu** [comment]
> Any particular training/inference code you have in mind for specific datasets, etc ?

No I don't. The model is pre-trained with ImageNet right? So we can inference with that. Here we have examples of training/inference various models: https://github.com/tensorflow/swift-models/tree/master/Examples. But I think this would not block merging your PR if you'd like to contribute something later!

**Comment by vballoli** [comment]
> > Any particular training/inference code you have in mind for specific datasets, etc ?
> 
> No I don't. The model is pre-trained with ImageNet right? 
This is just the model implementation and it's variants.
>So we can inference with that. Here we have examples of training/inference various models: https://github.com/tensorflow/swift-models/tree/master/Examples. 
I did think of an addition to this, but since all the image classification datasets are covered,  I wasn't sure where the ShuffleNetv2 model might fit.

**Comment by xihui-wu** [comment]
> > No I don't. The model is pre-trained with ImageNet right?
> 
> This is just the model implementation and it's variants.
> 
> > So we can inference with that. Here we have examples of training/inference various models: https://github.com/tensorflow/swift-models/tree/master/Examples.
> 
> I did think of an addition to this, but since all the image classification datasets are covered, I wasn't sure where the ShuffleNetv2 model might fit.

Imagenette could be the one. 
Can you remove swift-protobuf-prefix/src/swift-protobuf?

**Comment by xihui-wu** [comment]
> > > No I don't. The model is pre-trained with ImageNet right?
> > 
> > 
> > This is just the model implementation and it's variants.
> > > So we can inference with that. Here we have examples of training/inference various models: https://github.com/tensorflow/swift-models/tree/master/Examples.
> > 
> > 
> > I did think of an addition to this, but since all the image classification datasets are covered, I wasn't sure where the ShuffleNetv2 model might fit.
> 
> Imagenette could be the one.
> Can you remove swift-protobuf-prefix/src/swift-protobuf?

Bump. @vballoli your PR is pretty much ready to merge. Just can you remove swift-protobuf-prefix/src/swift-protobuf ?
